Project Pathfinder/Assets/Scripts/RenderMaze.cs

```diff
@@ -0,0 +1,96 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class RenderMaze : MonoBehaviour
+{
+    // Initialize fields on the inspector
+    [SerializeField]
+    [Range(1, 50)]
+    private int mazeWidth = 10;
+
+    [SerializeField]
+    [Range(1, 50)]
+    private int mazeHeight = 10;
+
+    [SerializeField]
+    private float cellSize = 1f;
+
+    [SerializeField]
+    private Transform wallPrefab = null;
+
+    [SerializeField]
+    private Transform floorPrefab = null;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        // Get the generated maze data
+        WallStatus[,] mazeData = GenerateMaze.Generate(mazeWidth, mazeHeight);
+
+        // Render the maze in the scene
+        Render(mazeData);
+    }
+
+    // Render the complete maze within the scene
+    private void Render(WallStatus[,] mazeData){
+        WallStatus currentCell = new WallStatus(); // Current maze cell being rendered
+        Vector2 scenePosition  = new Vector2();    // x,y position in the scene
+        var mazeFloor          = Instantiate(floorPrefab, transform);
+                                                   // Maze floor prefab
+
+        // Render the maze floor
+        mazeFloor.localScale = new Vector2(cellSize * (mazeWidth), cellSize * (mazeHeight));
+
+        // Render the cell walls of every maze cell
+        for (int j = 0; j < mazeHeight; j++){
+            for (int i = 0; i < mazeWidth; i++){
+                currentCell = mazeData[i,j];
+                scenePosition = new Vector2(cellSize * (-mazeWidth / 2 + i), cellSize * (-mazeHeight / 2 + j));
+
+                // Render the top wall of a maze cell
+                if(currentCell.HasFlag(WallStatus.TOP)){
+                    var topWall        = Instantiate(wallPrefab, transform) as Transform;
```

Comment:
Not a suggested change but just a question. Where is the "transform" variable coming from? I don't see it ever being created.

Reply:
Transform is a unity engine class, it is used here to make sure that the scale, rotation, and position values are cloned along with the base object